altair/utils/_vegafusion_data.py

```diff
@@ -0,0 +1,189 @@
+from typing import TypedDict, Union, Dict, Set, MutableMapping, Final
+
+from toolz import curried
+import uuid
+from weakref import WeakValueDictionary
+
+from altair.utils.core import _DataFrameLike
+from altair.utils.data import _DataType, _ToValuesReturnType, MaxRowsError
+from altair.vegalite.data import default_data_transformer
+
+# Temporary storage for dataframes that have been extracted
+# from charts by the vegafusion data transformer. Use a WeakValueDictionary
+# rather than a dict so that the Python interpreter is free to garbage
+# collect the stored DataFrames.
+extracted_inline_tables: MutableMapping[str, _DataFrameLike] = WeakValueDictionary()
+
+# Special URL prefix that VegaFusion uses to denote that a
+# dataset in a Vega spec corresponds to an entry in the `inline_datasets`
+# kwarg of vf.runtime.pre_transform_spec().
+VEGAFUSION_PREFIX: Final = "table://"
```

Comment:
maybe rename this to `vf_source://`? Since you also prefix a DataFrame as `table_<x>`, [here](https://github.com/altair-viz/altair/pull/3094/commits/07f8081ee70ec756beeed02cd1d423f62bc9dc58#diff-4789a42d9187ced48479b5409346321a0d110b18d5163aea482d74557681b52aR37). This will keep space for the support for arrays/tensors/matrices once they will be introduced.

Reply:
This prefix is build into VegaFusion. But there is an alternative prefix supported that I can switch to `vegafusion+dataset://`.

Reply:
renamed in [3a5517b](https://github.com/altair-viz/altair/pull/3094/commits/3a5517b55da95a7749e15973031b3e2ec38bcbd0)